From 24e46a0574b0976bc43b80e810962f6a231eeb58 Mon Sep 17 00:00:00 2001
Message-Id: <24e46a0574b0976bc43b80e810962f6a231eeb58.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Tue, 13 Jan 2015 13:44:49 +0530
Subject: [PATCH 13/29] MUST_REBASE [VPG]: video/adf: Add DP and EDP interface
 types

This patch adds DP and EDP as new interface types in ADF
and in Intel ADF layer.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5213
Change-Id: I7ceb1a2a6137653112c624de9e87d4c54b7a9b40
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/video/adf/adf.c                            |    2 +-
 drivers/video/adf/intel/core/intel_dc_config.c     |    5 ++++-
 .../video/adf/intel/include/core/intel_dc_config.h |    2 ++
 drivers/video/adf/intel/intel_adf_interface.c      |    4 ++++
 include/uapi/video/adf.h                           |    2 +-
 5 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/drivers/video/adf/adf.c b/drivers/video/adf/adf.c
index d5bfe97..75687fb 100644
--- a/drivers/video/adf/adf.c
+++ b/drivers/video/adf/adf.c
@@ -957,7 +957,7 @@ const char *adf_interface_type_str(struct adf_interface *intf)
 	case ADF_INTF_DSI:
 		return "DSI";
 
-	case ADF_INTF_eDP:
+	case ADF_INTF_EDP:
 		return "eDP";
 
 	case ADF_INTF_DPI:
diff --git a/drivers/video/adf/intel/core/intel_dc_config.c b/drivers/video/adf/intel/core/intel_dc_config.c
index af0c96a..4a558e5 100644
--- a/drivers/video/adf/intel/core/intel_dc_config.c
+++ b/drivers/video/adf/intel/core/intel_dc_config.c
@@ -69,7 +69,10 @@ static int intel_plane_validate(const struct intel_plane *plane)
 static int intel_pipe_validate(const struct intel_pipe *pipe)
 {
 	/*verify pipe type*/
-	if (pipe->type != INTEL_PIPE_DSI && pipe->type != INTEL_PIPE_HDMI) {
+	if (pipe->type != INTEL_PIPE_DSI &&
+	    pipe->type != INTEL_PIPE_HDMI &&
+	    pipe->type != INTEL_PIPE_DP &&
+	    pipe->type != INTEL_PIPE_EDP) {
 		dev_err(pipe->base.dev, "%s: Invalid pipe type %d\n", __func__,
 			pipe->type);
 		return -EINVAL;
diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index 203ce74..9bc1b14 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -221,6 +221,8 @@ struct intel_plane {
 enum intel_pipe_type {
 	INTEL_PIPE_DSI,
 	INTEL_PIPE_HDMI,
+	INTEL_PIPE_EDP,
+	INTEL_PIPE_DP,
 };
 
 enum intel_pipe_event {
diff --git a/drivers/video/adf/intel/intel_adf_interface.c b/drivers/video/adf/intel/intel_adf_interface.c
index 4e848e1..d85796e 100644
--- a/drivers/video/adf/intel/intel_adf_interface.c
+++ b/drivers/video/adf/intel/intel_adf_interface.c
@@ -237,6 +237,10 @@ static inline enum adf_interface_type to_adf_interface_type(
 		return ADF_INTF_DSI;
 	case INTEL_PIPE_HDMI:
 		return ADF_INTF_HDMI;
+	case INTEL_PIPE_DP:
+		return ADF_INTF_DPI;
+	case INTEL_PIPE_EDP:
+		return ADF_INTF_EDP;
 	default:
 		return ADF_INTF_TYPE_MAX;
 	}
diff --git a/include/uapi/video/adf.h b/include/uapi/video/adf.h
index 53be15d..294aa00 100644
--- a/include/uapi/video/adf.h
+++ b/include/uapi/video/adf.h
@@ -26,7 +26,7 @@
 
 enum adf_interface_type {
 	ADF_INTF_DSI = 0,
-	ADF_INTF_eDP = 1,
+	ADF_INTF_EDP = 1,
 	ADF_INTF_DPI = 2,
 	ADF_INTF_VGA = 3,
 	ADF_INTF_DVI = 4,
-- 
1.7.9.5

